app/code/local/Dwyera/Pinpay/Model/Result.php

```diff
@@ -165,6 +165,32 @@ public function getResponseToken()
             throw new Dwyera_Pinpay_Model_ResponseParseException;
         }
     }
+    public function getCustomerToken()
+    {
+        if (isset($this->msgObj->response->token)) {
```

Comment:
@zainengineer I have made new change for this to check if the response and response->card is set or not in isResponseParamExists().